orderbook/src/main.rs

```diff
@@ -160,34 +160,41 @@ async fn main() {
         .expect("failed to create gas price estimator"),
     );
 
-    let unsupported_tokens = HashSet::from_iter(args.shared.unsupported_tokens);
+    let pair_providers =
+        pool_aggregating::pair_providers(&args.shared.baseline_sources, chain_id, &web3).await;
+
     let mut base_tokens = HashSet::from_iter(args.shared.base_tokens);
     // We should always use the native token as a base token.
     base_tokens.insert(native_token.address());
+    let mut allowed_tokens = args.shared.allowed_tokens;
+    allowed_tokens.extend(base_tokens.iter().copied());
+    let unsupported_tokens = args.shared.unsupported_tokens;
     assert!(
-        unsupported_tokens
-            .intersection(&base_tokens)
-            .collect::<HashSet<_>>()
-            .is_empty(),
-        "Base tokens include at least one unsupported token!"
+        allowed_tokens
+            .iter()
+            .all(|token| !unsupported_tokens.contains(token)),
+        "same token can't be both allowed and denied"
     );
 
-    let current_block_stream = current_block_stream(web3.clone()).await.unwrap();
+    let bad_token_detector = Arc::new(ListBasedDetector {
+        allow_list: allowed_tokens,
+        deny_list: unsupported_tokens,
+        strategy: UnknownTokenStrategy::Allow,
```

Comment:
Don't we wanna use the `UnknownTokenStrategy::Forward` by default?

Reply:
Forward means it has to forward unknown tokens to another BadTokenDetecting implementation. But this new doesn't exist in this PR yet. This to keep the deny-list behavior the same. I'll make another PR to add the new better bad token detector as the forward target.